### PR TITLE
NXP-27349: Simplify filter logic and improve the tests

### PR DESCRIPTION
**Pull request by covolution**

### CODE REVIEW

File: nuxeo-ai-pipes/src/main/java/org/nuxeo/ai/pipes/filters/FacetFilter.java [hunk 1]
```diff
@@ -20,32 +20,42 @@
...
-    protected List<String> excludedFacets;
+    protected List<String> facets;
+
+    protected boolean excluded;
 
     @Override
     public void init(Map<String, String> options) {
-        includedFacets = propsList(options.get("includedFacets"));
-        excludedFacets = propsList(options.getOrDefault("excludedFacets", "Folderish"));
+        facets = propsList(options.get("includedFacets"));
```
Comment (PedroCardoso) [inline nuxeo-ai-pipes/src/main/java/org/nuxeo/ai/pipes/filters/FacetFilter.java]: never checked if null

File: nuxeo-ai-pipes/src/main/java/org/nuxeo/ai/pipes/filters/FacetFilter.java [hunk 2]
```diff
@@ -20,32 +20,42 @@
...
+        List<String> excludedFacets = propsList(options.get("excludedFacets"));
+
+        if (excludedFacets != null && !excludedFacets.isEmpty()) {
```
Comment (PedroCardoso) [inline nuxeo-ai-pipes/src/main/java/org/nuxeo/ai/pipes/filters/FacetFilter.java]: excludedFacets takes priority over includedFacets, in the case the two are present. 
maybe add this to javadoc

File: nuxeo-ai-pipes/src/main/java/org/nuxeo/ai/pipes/filters/FacetFilter.java (hunk 1 above)
Comment (covolution) [reply nuxeo-ai-pipes/src/main/java/org/nuxeo/ai/pipes/filters/FacetFilter.java]: The `propsList` method never returns null.

Comment (PedroCardoso) [reply nuxeo-ai-pipes/src/main/java/org/nuxeo/ai/pipes/filters/FacetFilter.java]: but you are testing it for the excludedFacets.

**Comment by PedroCardoso** [comment]
👍

**Pull request merged**